src/components/TextInput/TextInput.tsx

```diff
@@ -243,23 +251,35 @@ const TextInput = ({
           debouncedChange(e);
         }}
         multiLineIsResizable={multiLineIsResizable}
-        ref={inputRef}
+        ref={internalInputRef}
         {...inputProps}
       />
-      {onClear && (
-        <StyledIconContainer onClick={handleClear} {...iconContainerProps}>
+      {(onClear || clearable) && (
+        <StyledIconContainer
+          onClick={
+            onClear
+              ? (e: SyntheticEvent<Element, Event>) => handleClear(e)
+              : () => {
+                  if (internalInputRef?.current?.value) {
+                    setInternalValue('');
+                    internalInputRef.current.value = '';
+                  }
+                }
```

Comment:
I think it might be more clear for someone reading the code if this logic was all handled in the `handleClear` function